fe/fe-core/src/main/java/com/starrocks/load/InsertOverwriteJobManager.java

```diff
@@ -0,0 +1,239 @@
+            if (partitionIds != null) {
+                partitionIds.removeAll(job.getOriginalTargetPartitionIds());
+                if (partitionIds.isEmpty()) {
+                    partitionsWithOverwrite.remove(job.getTargetTableId());
+                }
+            } else {
+                partitionsWithOverwrite.remove(job.getTargetTableId());
+            }
+            overwriteJobMap.remove(jobid);
+            jobNum--;
+            return true;
+        } catch (Exception e) {
+            LOG.warn("deregister overwrite job failed", e);
+            throw e;
+        } finally {
+            lock.writeLock().unlock();
+        }
+    }
+
+    public boolean hasRunningOverwriteJob(long tableId) {
+        lock.readLock().lock();
+        try {
+            return partitionsWithOverwrite.containsKey(tableId);
+        } finally {
+            lock.readLock().unlock();
+        }
+    }
+
+    public void replayCreateInsertOverwrite(CreateInsertOverwriteJobInfo jobInfo) {
+        InsertOverwriteJob insertOverwriteJob = new InsertOverwriteJob(jobInfo.getJobId(),
+                jobInfo.getDbId(), jobInfo.getTableId(), jobInfo.getTargetPartitionIds());
+        boolean registered = registerOverwriteJob(insertOverwriteJob);
+        if (!registered) {
+            LOG.warn("register insert overwrite job failed. jobId:{}", insertOverwriteJob.getJobId());
+            return;
+        }
+        if (runningJobs == null) {
+            runningJobs = Lists.newArrayList();
+        }
+        runningJobs.add(insertOverwriteJob);
+    }
+
+    public void replayInsertOverwriteStateChange(InsertOverwriteStateChangeInfo info) {
+        InsertOverwriteJob job = getInsertOverwriteJob(info.getJobId());
+        InsertOverwriteJobRunner runner = new InsertOverwriteJobRunner(job);
+        runner.replayStateChange(info);
+        if (job.isFinished()) {
+            deregisterOverwriteJob(job.getJobId());
+            if (runningJobs != null) {
+                runningJobs.remove(job);
+            }
+        }
+    }
+
+    public void cancelRunningJobs() {
+        // resubmit running insert overwrite jobs
+        if (!GlobalStateMgr.isCheckpointThread()) {
+            cancelJobExecutorService.submit(() -> {
+                try {
+                    // wait until serving catalog is ready
+                    while (!GlobalStateMgr.getServingState().isReady()) {
+                        try {
+                            // not return, but sleep a while. to avoid some thread with large running interval will
+                            // wait for a long time to start again.
+                            Thread.sleep(1000);
+                        } catch (InterruptedException e) {
+                            LOG.warn("InsertOverwriteJobManager runAfterCatalogReady interrupted exception.", e);
+                        }
+                    }
+                    if (runningJobs != null) {
```

Comment:
needs lock to protect running jobs access.

Reply:
runningJobs is only used for replay, so no need to add a lock